src/sync/Synchroniser.js

```diff
@@ -100,12 +107,49 @@ export class Synchroniser {
     if (!this.isInitialised()) throw new Error('Not yet initialised');
     // Using async/await here means that any errors thrown by push or pull
     // will be passed up as a rejection of the promise returned by synchronise
+    await this.checkServerCompatibility();
     await this.push();
     await this.pull();
     this.settings.set(SYNC_LAST_SUCCESS, formatDate(new Date(), 'dots'));
   }
 
   /**
+   * Send server app version and process response appropriately:
+   * - reponse.compatible === true should complete the promise happy :)
+   * - reponse.compatible === false should reject the promise sad :(
+   * -- In the above case, response.error should specify the highest compatible version number of
+   *    mobile
+   * - else reject the promise, server isn't handling this check at all and needs updating
+   */
+  async checkServerCompatibility() {
+    const appVersion = await getAppVersion();
+    const serverURL = this.settings.get(SYNC_URL);
+    const response = await fetch(
+      `${serverURL}/sync/v2/msupply_mobile_compatibility/?app_version=${appVersion}`,
```

Comment:
@craigdrown @nmadruga @andreievg Requested y'all to review particularly for agreement on the REST api structure as depicted here